changelogs/unreleased/5657-shadialtarsha-minor.md

```diff
@@ -0,0 +1,42 @@
+## Add support for extensions.filters.http.ratelimit.v3.RateLimitPerRoute 
+
+Envoy has extensions.filters.http.ratelimit.v3.RateLimitPerRoute API which allows control over the Vhost Rate Limits on the route level.
+
+### Sample Configurations
+#### httpproxy.yaml
+```yaml
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: echo
+spec:
+  virtualhost:
+    fqdn: local.projectcontour.io
+    rateLimitPolicy:
+      global:
+        disabled: true
```

Comment:
In hindsight it seems to be that the api here should be the same and a bit recursive.

I think it should have been:

```
    rateLimitPolicy:
        globalPolicy: "Ignore|include|Override"
 ```

Reply:
In general we have 3 layers:

* Global
* HTTProxy
* Route

And each one can be merged with the above policy but the APIs are not the same which makes things a bit confusing imho. We should aim for consistency here